pkg/didmethod/httpbinding/resolver.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package httpbinding
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	errors "golang.org/x/xerrors"
+)
+
+// HTTPClientOpts holds options for the HTTP transport implementation of DIDResolver
+// it has a required http.Client instance and optional basePath which is appended to the remove DID Resolver URL
+type HTTPClientOpts struct {
+	client   *http.Client
+	basePath string
+}
+
+// HTTPOptSetter is an outbound HTTP transport option
+type HTTPOptSetter func(opts *HTTPClientOpts)
+
+// WithHTTPClientInstance option is for creating an Outbound HTTP transport using an http.Client instance
+func WithHTTPClientInstance(client *http.Client) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client = client
+	}
+}
+
+// WithHTTPClientTimeout option is for creating an Outbound HTTP transport using a client timeout value
+func WithHTTPClientTimeout(timeout time.Duration) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client.Timeout = timeout
+	}
+}
+
+// WithHTTPClientTLSConfig option is for creating an Outbound HTTP transport using a tls.Config instance
+func WithHTTPClientTLSConfig(tlsConfig *tls.Config) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: tlsConfig,
+			},
+		}
+	}
+}
+
+// WithHTTPClientBasePath options is for setting optional base path, for example "document"
+func WithHTTPClientBasePath(basePath string) HTTPOptSetter {
```

Comment:
we don't need WithHTTPClientBasePath. You can move basePath to New

Reply:
This indicates that DIDs could be resolved using hostname like https://uniresolver.io/ directly or by appending extra relative path like "https://uniresolver.io/document/".
@troyronda what do you think about this?

Reply:
@fqutishat @kdimak I would likely not add a new basePath argument to New.